clusterloader2/pkg/measurement/util/pvcs.go

```diff
@@ -0,0 +1,129 @@
+/*
+Copyright 2019 The Kubernetes Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package util
+
+import (
+	"fmt"
+
+	corev1 "k8s.io/api/core/v1"
+	"k8s.io/apimachinery/pkg/util/sets"
+)
+
+// PVCStartupStatus represents phase of a pvc group.
+type PVCStartupStatus struct {
+	Pending  int
+	Bound    int
+	Lost     int
+	Expected int
+}
+
+// String returns string representation for PVCsStartupStatus.
+func (s *PVCStartupStatus) String() string {
+	return fmt.Sprintf("PVCs: %d out of %d bound, %d pending, %d lost", s.Bound, s.Expected, s.Pending, s.Lost)
+}
+
+// ComputePVCStartupStatus computes PVCsStartupStatus for a group of PVCs.
+func ComputePVCStartupStatus(pvcs []*corev1.PersistentVolumeClaim, expected int) PVCStartupStatus {
+	startupStatus := PVCStartupStatus{
+		Expected: expected,
+	}
+	for _, p := range pvcs {
+		if p.Status.Phase == corev1.ClaimPending {
+			startupStatus.Pending++
+		} else if p.Status.Phase == corev1.ClaimBound {
+			startupStatus.Bound++
+		} else if p.Status.Phase == corev1.ClaimLost {
+			startupStatus.Lost++
+		}
+	}
+	return startupStatus
+}
+
+type pvcInfo struct {
```

Comment:
I'm wondering if we need that.

The reason why we added that for pods is that pod may be running and turn into not-running after that.

@msau42 - Is it the same case for PVCs?
I mean, is it possible that claim went to Bound state and then it will go to Lost state?

If not, I would say we should remove it, as this whole logic for diffs, isn't really needed.
[For the context: we did this for pods to avoid silently skipping the situations that a lot of pods are silently dying, but because new ones are created by RS, we are eventually converging to the desired state.]

Reply:
It is possible if the underlying PV is lost. See: https://github.com/kubernetes/kubernetes/issues/60298#issuecomment-367997711

Reply:
Lost is a possible state for a PVC to be in, but I wouldn't expect that to happen in a normal test run.  Unless there is some bug (and in that case it would be good to catch too).

Reply:
Got it - thanks helps. Thanks!